apps/gro/controllers/address-start.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+const BaseController = require('hof').controllers.base;
+const Client = require('node-rest-client').Client;
+const config = require('../../../config');
+var logger = require('../../../lib/logger');
+
+module.exports = class AddressStartController extends BaseController {
+  constructor(options) {
+    super(options);
+  }
+
+  getValues(req, res) {
+    const postcode = req.sessionModel.attributes.postcode.toUpperCase();
+    req.sessionModel.set('postcode', postcode);
+    const previouspostcode = req.sessionModel.attributes.previouspostcode;
+    req.sessionModel.set('previous-step', res.locals.backLink);
+    if (postcode !== previouspostcode) {
+      const client = new Client();
+      const url = config.postcode.hostname + config.postcode.addresses.path;
+      const args = {
+        parameters: {},
+        headers: {'Authorization': config.postcode.authorization || ''}
```

Comment:
Will the service work without an authorization code? If not, then there's no point in this, perhaps better to throw an error if no authorization code has been set.